packaging/packager_deb.sh

```diff
@@ -5,46 +5,64 @@ readonly distro="${DISTRO:-ubuntu}"
 readonly distro_version="${VERSION:-xenial}"
 readonly distro_root="${ROOT:-$HOME}"
 
-# deb dependencies
+echo "------------- deb dependencies -------------"
 export DEBIAN_FRONTEND=noninteractive
 apt-get update
-# python3
-apt-get -y install python3 python3-coverage
-# python-imaging
-apt-get -y install python3-pil
-# pip dependencies (for dependencies not available as DEB)
+echo "------------- python3 install -------------"
+apt-get -y install python3 python3-venv
+echo "----- pip dependencies (for dependencies not available as DEB) -----"
 apt-get -y install gcc libx11-dev libxtst-dev python3-dev libpng-dev python3-pip
+
+echo "----- create a venv to use that as the python env (needed for python 3.12 compatibility specific to Ubuntu) -----"
+python3 -m venv docker_venv --system-site-packages
+source docker_venv/bin/activate
+
+echo "------------- upgrade pip -------------"
 pip3 install --upgrade pip
-# contour, template, feature, cascade, text matching
+
+echo "------------ install coverage ------------"
+pip3 install coverage
+echo "------------- python-imaging -------------"
+apt-get -y install python3-pil
+
+echo "------------- contour, template, feature, cascade, matching -------------"
 apt-get -y install python3-numpy
 if [[ $distro_version == "xenial" ]]; then
     export DISABLE_OPENCV=1
 else
     apt-get -y install python3-opencv
 fi
-# text matching
+
```

Comment:
Are we following specific logic which blocks to separate and which not? I originally thought that blocks like "matching deps" and "debian deps" were separated by an empty line and combine subblocks without these empty lines but now i am confused about the separation of text matching from the rest of the matching above. Could we either put these together or simply put an empty line before each "echo" heading for simplicity?

Reply:
I'm putting an empty line on each echo